lib/src/component_declaration/flux_component.dart

```diff
@@ -228,24 +229,40 @@ abstract class FluxUiStatefulComponent2<TProps extends FluxUiProps, TState exten
 ///
 /// Private so it will only get used in this file, since having lifecycle methods in a mixin is risky.
 mixin _FluxComponentMixin<TProps extends FluxUiProps> on component_base.UiComponent<TProps>, BatchedRedraws {
-  static final Logger _logger = Logger('over_react._FluxComponentMixin');
-
   /// List of store subscriptions created when the component mounts.
   ///
   /// These subscriptions are canceled when the component is unmounted.
   List<StreamSubscription> _subscriptions = [];
 
   void _validateStoreDisposalState(Store store) {
-    String message = 'Cannot listen to a disposed/disposing Store.';
+    if (store.isOrWillBeDisposed) {
```

Comment:
I wrapped both the assert and the log in this check so we could conditionally build a shared message instead of computing it unnecessarily. The control-flow of this method should be the same as it was before.